lib/moxinet/plug/mocked_response.ex

```diff
@@ -77,8 +77,13 @@ defmodule Moxinet.Plug.MockedResponse do
           callback.(nil)
 
         _other ->
-          body = if json_body?(conn), do: Jason.decode!(body), else: nil
-          callback.(body)
+          if json_body?(conn) do
+            callback.(Jason.decode!(body))
+          else
+            # We still want to pass a request body to the expect function for other
+            # content types, for example application/x-www-form-urlencoded
+            callback.(body)
+          end
```

Comment:
```suggestion
          body = 
            if json_body?(conn) do
              Jason.decode!(body)
            else
              body
            end
          
          callback.(body)
```

Perhaps this makes it clearer that we always want to pass a body but want to give special treatment to it if it's JSON?